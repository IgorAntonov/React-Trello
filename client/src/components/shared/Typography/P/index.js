import styled from 'styled-components';

import { media } from '../../helpers';

export const P = styled.p`
  margin-top: 0;
  margin-bottom: 0.5rem;

  font-size: 0.8rem;
  font-weight: 400;

  ${media.mobile`
    font-size: 0.8rem;
  `}
  ${media.tablet`
    font-size: 0.9rem;
  `}
  ${media.desktop`
    font-size: 1rem;
  `}
`;
