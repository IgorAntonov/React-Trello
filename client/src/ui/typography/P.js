import styled from 'styled-components';
import PropTypes from 'prop-types';

import { media } from 'Src/helpers';

export const P = styled.p`
  margin-top: 0;
  margin-bottom: ${({ marginB }) => marginB || '0.5rem'};
  
  color: ${({ color }) => color || '#fafafa'};
  background-color: ${({ bgcolor }) => bgcolor || 'transparent'};
  
  cursor: default;
  font-size: 0.8rem;
  font-weight: 400;
  user-select: none;

  ${media.mobile`
    font-size: 0.8rem;
  `}
  ${media.tablet`
    font-size: 0.9rem;
  `}
  ${media.desktop`
    font-size: 1rem;
  `}
`;
P.propTypes = {
  marginB: PropTypes.string,
  color: PropTypes.string,
  bgcolor: PropTypes.string
};
