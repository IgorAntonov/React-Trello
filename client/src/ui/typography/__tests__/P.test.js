import React from 'react';
import { shallow } from 'enzyme';

import { P } from '../P';

describe('<P/>', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<P />);
  });

  test('renders', () => {
    expect(wrapper.exists()).toBe(true);
  });
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
