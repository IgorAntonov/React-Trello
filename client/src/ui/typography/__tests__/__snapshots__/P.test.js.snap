// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<P/> should match snapshot 1`] = `
<p
  className="P-gHPbqS dwmwDC"
/>
`;
